### PR TITLE
Fix with_H0/little-h equivalency

### LINKED ISSUE #8271: Problem with the `littleh` part of unit equivalencies?
In the newly added `littleh` equivalencies: http://docs.astropy.org/en/stable/units/equivalencies.html#unit-equivalencies 

We notice that the implementation of `littleh` seems to be wrong, as highlighted in the following figure:

screen shot 2018-12-12 at 12 59 23

If `distance = 100 Mpc/h`, and `h=0.7`, should it be equivalent to 140 Mpc, instead of 70Mpc? 

I can reproduce this so it is not a typo...

### PR DESCRIPTION
**Pull request by eteq**
This should fix #8271. 

A double-check by @dr-guangtou and/or @aphearin would be greatly appreciated, given that I already messed this up once! (especially the magnitudes, although admittedly that's a bit more ambiguous because it really depends on how it's expressed in a paper to be sure which is the "right way")

### CODE REVIEW

**Comment by pllim** [comment]
Need to fix one remaining `doctest`:
```
_______________________ [doctest] docs/whatsnew/3.1.rst ________________________
312 conversion between physical units and so called "little-h" units, a frequent
313 source of confusion for novice (and not-so-novice...) extragalactic astronomers
314 and cosmologists.  To see it in action::
315 
316   >>> import astropy.units as u
317   >>> from astropy.cosmology import WMAP9
318   >>> distance = 100 * (u.Mpc/u.littleh)
319   >>> distance  # doctest: +FLOAT_CMP
320   <Quantity 100. Mpc / littleh>
321   >>> distance.to(u.Mpc, u.with_H0(WMAP9.H0))  # doctest: +FLOAT_CMP
Expected:
    <Quantity 69.32 Mpc>
Got:
    <Quantity 144.25851125 Mpc>
```

Also please add a change log. Thanks!

**Comment by eteq** [comment]
Thanks @pllim - tests are now passing!

**Comment by eteq** [comment]
@dr-guangtou - given https://github.com/astropy/astropy/issues/8271#issuecomment-447537586 feel free to also have Peter B have a look at this... (although perhaps that's beyond-the-pale if he doesn't have a github account with which to do the review :wink:)

File: CHANGES.rst [hunk 1]
```diff
@@ -313,6 +313,9 @@ astropy.units
 
 - Ensure correctness of units when raising to a negative power. [#8263]
 
+- Fix ``with_H0`` equivalency to how the appropriate direction of
```
Comment (mhvk) [inline CHANGES.rst]: unclear grammar

File: astropy/units/equivalencies.py [hunk 2]
```diff
@@ -728,6 +728,6 @@ def with_H0(H0=None):
         from astropy import cosmology
         H0 = cosmology.default_cosmology.get().H0
 
-    h100_val_unit = Unit(H0.to((si.km/si.s)/astrophys.Mpc).value/100 * astrophys.littleh)
+    h100_val_unit = Unit(100/(H0.to((si.km/si.s)/astrophys.Mpc).value) * astrophys.littleh)
```
Comment (mhvk) [inline astropy/units/equivalencies.py]: While updating this, might as well use `H0.to_value(unit)` (rather than `.to(unit).value`, which is slower).

**Comment by aphearin** [comment]
The two checks on code-correctness I have done are:
1. When converting distances from h=1 units to h=0.7 units, the numerical value _increases_. 
2. When converting magnitudes from h=1 units to h=0.7 units, the numerical value _decreases_. 

Both the code and the documentation now reflect the correct sign of the behavior.

**Pull request merged**